proposals/interface-types/working-notes/choices/webidl.md

```diff
@@ -0,0 +1,270 @@
+_algebraic type definition_. This is, in part, to regularize concepts such as
+nullability and to permit limited forms of polymorphism (allowing a color to be
+named by a string, a vector of small integers or a large integer, for example).
+
+WebIDL explicitly prohibits the formation of new types; indeed it uses type
+unions to achieve similar polymorphism. However, as we note above, not all
+programming languages support dynamic types or type unions; which makes it
+problematic to support such polymorphism.
+
+The form of algebraic types envisaged in the Interface Types system is
+intentionally limited. For example, we do not permit _recursive_ types to be
+defined. This is in keeping with the vision of supporting APIs across ownership
+boundaries where complex data is likely to take limited form.[^This design
+choice may be revisited in future versions of the Interface Types specification
+if supporting structured types such as abstract XML and/or JSON become
+important.]
+
+### Working with WebIDL
+
+While the Interface Types language introduces its own IDL, it is still clearly
+important to support WebIDL; particularly for Web APIs.
+
+The strategy that we adopt can be summarized as interpreting WebIDL as an
+alternate target to WebAssembly itself.
+
+When using Interface Types to specify a public API for a WebAssembly module, the
+author must provide special _adapters_ -- import adapters for accessing external
+APIs and export adapters when offering a public API.
+
+Typically, such adapters would be automatically generated by language
+specific tools; which would mean that individual code authors would likely be
+not involved in the construction of the adapters.
+
+Similarly, in the case where a WebIDL API is involved, there would be adapters
+between the Interface Types language and WebIDL. There would also be adapters
+between WebIDL and the host environment; although these are usually private to
+the host.
+
+In case case of adapting between WebIDL and Interface Types, it is hoped that a
+single method for automatically constructing adapters would be sufficient; and
+automatic. This is should be possible, based on the fact that the Interface
+Types language has been designed to be a slight extension of a simplification of
+WebIDL.
+
+Note that this does not imply that the semantics of Interface Types and WebIDL
+are completely aligned: any automatically generated adapters would likely ignore
+those aspects of WebIDL that are not accounted for in the Interface Types
+language.
+
+#### Modifying WebIDL itself
+
+One of the strategies that we considered was modifying WebIDL itself. After all, as we note, the Interface Types language is somewhat based on WebIDL.
+
+However, for some of the same reasons outlined here, changing WebIDL is
+considered to be infeasible: WebIDL has an orientation to JavaScript that would
+be difficult to carry forward in such a refactoring.
+
+## Summary
```

Comment:
I think the rough structure of the argument against using Web IDL is as laid out in [this slide](https://docs.google.com/presentation/d/1wtAknL-UJWDoIgSbyF5paTBSpVVj-fKU4tiHMxJbSzE); viz., that, to use Web IDL we'd need to:
1. remove a bunch of stuff that didn't make sense for wasm
2. add a bit of stuff that only makes sense for wasm

And now everyone's confused when we're saying "Web IDL", and so it'd probably be less work with a better end result if we just define a new thing and define how it maps onto existing WebIDL-defined Web APIs.